agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -4319,6 +4327,89 @@ bool backhaul_manager::add_link_metrics(const sMacAddr &reporter_al_mac,
 
     return true;
 }
+
+bool backhaul_manager::handle_channel_selection_request(ieee1905_1::CmduMessageRx &cmdu_rx,
+                                                        const std::string &src_mac)
+{
+    const auto mid = cmdu_rx.getMessageId();
+
+    LOG(DEBUG) << "Forwarding CHANNEL_SELECTION_REQUEST to son_slave, mid=" << std::hex << int(mid);
+    m_expected_channel_selection.mid = mid;
```

Comment:
As @mariomaz said, there should be something like this:

```suggestion
    // Clear previous request, if any
    m_expected_channel_selection.requests.clear();
    m_expected_channel_selection.responses.clear();
    
    m_expected_channel_selection.mid = mid;
```

Reply:
added proposed fix into the code